### DISCUSSION TITLE
I don't understand how customer group discounts work (read the description)

### DISCUSSION THREAD

**Discussion by xV4L3x** [lunarphp/lunar#1546] [Q&A]
Steps I made:

- Created a customer group from the admin panel.
- Created a discount from the admin panel and assigned only to the new customer group.
- Logged in with an user which is linked to a customer that belongs to the new customer group.
- In the checkout process I added the discount coupon code to the cart and made a refresh()

No discount is applying, I also tried to link the customer to the cart:
`$this->cart->setCustomer($this->user->latestCustomer());`

What am I doing wrong?

**Comment by glennjacobs** [comment]
Difficult to tell with limited information. Have you assigned the correct channel to the discount?

**Reply by xV4L3x** [reply]
Yes I have two customer groups, retail and premium, I only enabled the discount for the premium group, the products in the store are purchasable by both the premium and retail group.

**Reply by glennjacobs** [reply]
I said Channel though...

**Reply by xV4L3x** [reply]
I'm sorry I misunderstood your question, by the way, yes I only have the webstore as a channel and the discount is assigned to it

**Comment by alecritson** [comment]
Are you recalculating the cart after saving the discount code to it? Can you post a dump of what your cart looks like afterwards. Also any DB screenshots/dumps of the set up discounts would be useful.

**Reply by xV4L3x** [reply]
this it how I am handling the discount process, I managed to workaround the validity of the discount checking the amount of discount applied, but a more elegant solution is appreciated:

```php
public function applyDiscount() {

        /**
         * Here the implementation verifies the validity of the
         * discount code only whenever the actual discount applied is > 0
         */

        //check if discount code is valid
        $discount = Discount::active()
            ->usable()
            ->where('coupon',  $this->discountCode)->first();


        if (!$discount) {
            $this->addError('discountCode', 'Invalid discount code');
            return;
        }


        //apply the coupon and recalculate
        $this->applyCoupon($discount);

        //check if discount is valid watching if the discount is greater than 0
        if ($this->cart->discountTotal->value > 0) {
            $this->currentStep = $this->steps['payment'];
        } else {
            $this->addError('discountCode', 'Invalid discount code');
            $this->removeDiscount();
            return;
        }

        $this->currentStep = $this->steps['payment'];
 }
```
    
    Note, before calling the function the logged user is linked to the customer  and applied to the cart:
    
```php
//load user
        $this->user = auth()->user();

        //get latest customer
        if ($this->user) {
            $customer = $this->user->latestCustomer();
            if ($customer) {
                $this->cart->setCustomer($customer);
                $this->cart->save();
                $this->refreshCart();
            }
        }
```
